### PR TITLE
HRINT-356 Java Demos - Static Indexes

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/mapReduceIndex/metadata.json [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            "DescriptionLines": [
+                "* The `reduce` function is the second stage.<br>",
+                "  It operates on the Map results, specifying how the data should be _grouped_ and _aggregated_.",
+                "",
+                "* Specify the field to group-by:<br><br>",
+                "  `GroupBy(result => result.Country)`  ==> group by _index-field_ 'Country'",
+                "",
+                "* Specify the aggregation needed on that field:<br><br>",
+                "  `Country = g.Key`  ==>  The country we group by<br>",
+                "  `CountryCount = Enumerable.Sum(g, x => ((int) x.CountryCount))`  ==> The aggregating function"
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/mapReduceIndex/metadata.json]: can we check if this cast to int is needed? less is better. :)

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/StaticIndexesOverview.java [hunk 2]
```diff
@@ -50,7 +51,7 @@ public void run() {
         try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
             //region Step_5
             IDocumentQuery<Employee> queryOnIndex = session.query(Employees_ByLastName.Result.class, Employees_ByLastName.class)
-                .whereEquals("LastName", "SomeNAme")
+                .whereEquals("LastName", "SomeName")
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/StaticIndexesOverview.java]: thanks :)

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        {
+            "Title": "The index Result class",
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* The Result class defines the _index-fields_ - the fields that will be indexed.<br>",
+                "  Later on, a query can be made on these index-fields.",
+                "",
+                "* The _content_ for each index-field is configured in the index constructor (see next step).",                
+                "",
+                "* In the above index, field 'lastName' is indexed and will be queryable."
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json]: LastName?

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+                "* _Additional indexing options, per index-field, that can be defined inside the Map function are_:<br>",
+                "   Index spatial data<br>",
+                "   Index dynamic fields<br>",
+                "   Index hierarchical data<br>",
+                "   Index document fields as json<br>",
+                "   Boosting the indexed field value<br>",
+                "   Index counters & attachments names<br>",
+                "   Index data from a related document<br>",
+                "   Index data from the document's metadata<br>",
+                "   Use LINQ extensions, such as: reverse, parseint, and more",
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json]: I think extensions names are cased inproperly ?

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json (hunk 3 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json]: This text is supposed to explain the 'result' class (which contains 'lastName')
Wouldn't it be confusing to have 'LastName' ?

Comment (ml054) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json]: no, index doesn't index result classes. Index works on documents. And document has field LastName. 

So, it goes like this: field LastName from JSON document is indexed, then you can query LastName field and assign to field lastName in result class.

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json (hunk 4 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/staticIndexesOverview/metadata.json]: fixed

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/mapReduceIndex/metadata.json (hunk 1 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/staticIndexes/mapReduceIndex/metadata.json]: not needed, works w/o

**Pull request merged**